### PR TITLE
Add _get_type_id to Impl

### PR DESCRIPTION
**Pull request by maksimdrachov**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new method for type identification in the event handling framework, enhancing the functionality of the `EventProxy` class.

### CODE REVIEW

File: include/olg_scheduler/olg_scheduler.hpp [hunk 1]
```diff
@@ -97,6 +97,11 @@ class Event : private cavl::Node<Event<TimePoint>>
     /// It is guaranteed that while an event resides in the tree, it has a valid deadline set.
     [[nodiscard]] std::optional<TimePoint> getDeadline() const noexcept { return deadline_; }
 
+    static constexpr std::array<std::uint8_t, 16> _get_type_id_() noexcept
```
Comment (pavel-kirienko) [inline include/olg_scheduler/olg_scheduler.hpp:101]: This thing needs a comment explaining that it is added for the benefit of cetl::unbounded_variant

Comment (maksimdrachov) [reply include/olg_scheduler/olg_scheduler.hpp:101]: Added.

**Review by serges147 — APPROVED** [review]

**Comment by serges147** [comment]
@maksimdrachov Maxim, when I reviewed I forgot about three things:
1. Let's add a simple unit test(s) - just to make sure that newly exposed public `_get_type_id_` is available and returning our expected value. Ideally it would test it with `cetl::unbounded_variant` as well but I understand extra complexities by bringing CETL dependency...
2. There is `CHANGELOG.md` file which normally should be updated, but see bullet # 3...
3. If we start using OLG (in EPM, right?) then probably it makes sense to "release olg", at least by making a tag (f.e. as `0.2.0`)... Though, we can do this later as well - especially if we still need prove that on real project (like EPM is) we don't have any problems with OLG and everything works fine.
Thanks!

**Comment by maksimdrachov** [comment]
Regarding `2)` and `3)`, I'm currently stuck debugging a strange CAN issue, once that is addressed I can merge the [MR](https://github.com/Zubax/fluxgrip_software/pull/354) that switches the firmware to olg_scheduler. I've added a TODO to that MR.

Regarding 1), I opened an [MR](https://github.com/Zubax/olg_scheduler/pull/9) to remind myself to do this.

**Pull request merged**